climada_petals/hazard/tc_rainfield.py

```diff
@@ -66,204 +215,1562 @@ def set_from_tracks(self, *args, **kwargs):
+                                              shape=rainrates_shape)
+        rainrates_sparse.eliminate_zeros()
+
+    return intensity_sparse, rainrates_sparse
+
+def _compute_rain_sparse_chunked(
+    track_centr_msk: np.ndarray,
+    track: xr.Dataset,
+    *args,
+    max_memory_gb: float = DEF_MAX_MEMORY_GB,
+    **kwargs,
+) -> Tuple[sparse.csr_matrix, Optional[sparse.csr_matrix]]:
+    """Call `_compute_rain_sparse` for chunks of the track and re-assemble the results
 
-def _RCLIPER(fmaxwind_kn, inreach, radius_km):
-    """Calculate rainrate in mm/h based on RCLIPER given windspeed (kn) at
-    a specific node
     Parameters
     ----------
-    fmaxwind_kn : float
-        maximum sustained wind at specific node
-    inreach : np.array, boolean
-        1 if centroid is within dist_degree,
-        0 otherwise
-    radius_km : np.array
-        distance to node for every centroid
+    track_centr_msk : np.ndarray
+        Each row is a mask that indicates the centroids within reach for one track position.
+    track : xr.Dataset
+        Single tropical cyclone track.
+    max_memory_gb : float, optional
+        Maximum memory requirements (in GB) for the computation of a single chunk of the track.
+        Default: 8
+    args, kwargs :
+        The remaining arguments are passed on to `_compute_rain_sparse`.
+
+    Returns
+    -------
+    intensity, rainrates :
+        See `_compute_rain_sparse` for a description of the return values.
     """
+    npositions = track.sizes["time"]
+    # The memory requirements for each track position are estimated for the case of 25 arrays
+    # containing `nreachable` float64 (8 Byte) values each. The chunking is only relevant in
+    # extreme cases with a very high temporal and/or spatial resolution.
+    max_nreachable = max_memory_gb * 1e9 / (8 * 25 * npositions)
+    split_pos = [0]
+    chunk_size = 3
+    while split_pos[-1] + chunk_size < npositions:
+        chunk_size += 1
+        # create overlap between consecutive chunks
+        chunk_start = max(0, split_pos[-1] - 2)
+        chunk_end = chunk_start + chunk_size
+        nreachable = track_centr_msk[chunk_start:chunk_end].any(axis=0).sum()
+        if nreachable > max_nreachable:
+            split_pos.append(chunk_end - 1)
+            chunk_size = 3
+    split_pos.append(npositions)
 
-    rainrate = np.zeros(len(inreach))
+    intensity = []
+    rainrates = []
+    for prev_chunk_end, chunk_end in zip(split_pos[:-1], split_pos[1:]):
+        chunk_start = max(0, prev_chunk_end - 2)
+        inten, rainr = _compute_rain_sparse(
+            track.isel(time=slice(chunk_start, chunk_end)), *args,
+            max_memory_gb=max_memory_gb, **kwargs,
+        )
+        intensity.append(inten)
+        rainrates.append(rainr)
 
-    # Define Coefficients (CLIPER NHC bias adjusted (Tuleya, 2007))
-    a1 = -1.1  # inch per day
-    a2 = -1.6  # inch per day
-    a3 = 64.   # km
```

Comment:
There was a typo in this line that is fixed in this PR. The value for a3 is 64.5 instead of 64.0, see Table 2 in Tuleya et al. 2007:
![image](https://github.com/CLIMADA-project/climada_petals/assets/57705593/385e5ef3-391b-478d-8d9c-9345c79f1909)
That's why the intensity values in the unit tests have changed.

Reply:
good, tahnks. I think it was me implementing the original R-CLIPER, but I mostly translated the CLIMADA MATLAB code without looking into the refs.